connectors/http/polling/src/main/java/io/camunda/connector/http/polling/HttpPollingConnector.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.http.polling;
+
+import io.camunda.connector.api.annotation.InboundConnector;
+import io.camunda.connector.api.inbound.InboundConnectorExecutable;
+import io.camunda.connector.api.inbound.InboundIntermediateConnectorContext;
+import io.camunda.connector.feel.ConnectorsObjectMapperSupplier;
+import io.camunda.connector.http.base.components.HttpTransportComponentSupplier;
+import io.camunda.connector.http.base.services.HttpService;
+import io.camunda.connector.http.polling.model.PollingIntervalConfiguration;
+import io.camunda.connector.http.polling.service.HttpPollingSharedExecutorService;
+import io.camunda.connector.http.polling.task.HttpRequestTask;
+import io.camunda.connector.http.polling.task.PollingOperateTask;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@InboundConnector(name = "HTTP_POLLING", type = "io.camunda:http-polling:1")
+public class HttpPollingConnector
+    implements InboundConnectorExecutable<InboundIntermediateConnectorContext> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(HttpPollingConnector.class);
+
+  private final HttpService httpService;
+  private final HttpPollingSharedExecutorService executorService;
+
+  public HttpPollingConnector() {
+    this(null);
+  }
+
+  public HttpPollingConnector(final String proxyFunctionUrl) {
+    this(
+        new HttpService(
+            ConnectorsObjectMapperSupplier.getCopy(),
+            HttpTransportComponentSupplier.httpRequestFactoryInstance(),
+            proxyFunctionUrl),
+        HttpPollingSharedExecutorService.getInstance());
+  }
+
+  public HttpPollingConnector(
+      final HttpService httpService, final HttpPollingSharedExecutorService executorService) {
+    this.httpService = httpService;
+    this.executorService = executorService;
+  }
+
+  @Override
+  public void activate(final InboundIntermediateConnectorContext context) {
+    PollingIntervalConfiguration intervals =
+        context.bindProperties(PollingIntervalConfiguration.class);
+
+    HttpRequestTask httpRequestTask = new HttpRequestTask(httpService, context::correlate);
+    executorService.scheduleWithFixedDelay(
+        httpRequestTask, 0, intervals.getHttpRequestInterval().toMillis(), TimeUnit.MILLISECONDS);
+
+    PollingOperateTask pollingOperateTask = new PollingOperateTask(context, httpRequestTask);
+    executorService.scheduleWithFixedDelay(
+        pollingOperateTask,
+        0,
+        intervals.getOperatePollingInterval().toMillis(),
+        TimeUnit.MILLISECONDS);
+  }
```

Comment:
Why do you schedule two tasks here?

Shouldnt we only schedule the polling operate task and if the process instance is not already scheduled for polling create the HttpPolling task and schedule that one?

Reply:
Sure, its understood but this will for example run each individual http request sequentially. Please implement the following:
1. Request process instances from Operate
2. If they are not scheduled yet, request variables and schedule them
3. Each individuall http request will get executed with the delay configures as the request interval (not the operate polling interval)
4. Remove process instances from scheduling that were not fateched from operate (they finished already or were cancelled because of a boundary event)